modules/programs/streamlink.nix

```diff
@@ -0,0 +1,143 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.streamlink;
+
+  renderSettings = mapAttrsToList (name: value:
+    if (builtins.isBool value) then
+      if value then name else ""
+    else if (builtins.isList value) then
+      concatStringsSep "\n"
+      (builtins.map (item: "${name}=${builtins.toString item}") value)
+    else
+      "${name}=${builtins.toString value}");
```

Comment:
This looks like it's re-implementing [lib.generators.toKeyValue](https://github.com/nixos/nixpkgs/blob/master/lib/generators.nix#L174-L174) ? Similar to https://github.com/nix-community/home-manager/pull/6674

Reply:
`toKeyValue` can't handle option key without value (the `=` separator will always render). You can see `player-no-close` option in the test case for an example.

Reply:
I can, of course, tweak the generator `mkKeyValue` function, but it feels just doing the same thing as writing my own function here.